src/SFA.DAS.Reservations.Application.UnitTests/AccountReservation/Services/WhenChangingPartyOfAReservation.cs

```diff
@@ -0,0 +1,124 @@
+﻿using System;
+using System.Threading.Tasks;
+using AutoFixture.NUnit3;
+using FluentAssertions;
+using Moq;
+using NUnit.Framework;
+using SFA.DAS.Reservations.Application.AccountReservations.Services;
+using SFA.DAS.Reservations.Domain.AccountLegalEntities;
+using SFA.DAS.Reservations.Domain.Exceptions;
+using SFA.DAS.Reservations.Domain.Reservations;
+using SFA.DAS.Testing.AutoFixture;
+
+namespace SFA.DAS.Reservations.Application.UnitTests.AccountReservation.Services
+{
+    public class WhenChangingPartyOfAReservation
+    {
+        [Test, MoqAutoData]
+        public void And_Reservation_Not_Found_Then_Throws_EntityNotFoundException(
+            ChangeOfPartyServiceRequest request,
+            [Frozen] Mock<IReservationRepository> mockRepository,
+            AccountReservationService service)
+        {
+            mockRepository
+                .Setup(repository => repository.GetById(request.ReservationId))
+                .ReturnsAsync((Domain.Entities.Reservation) null);
+
+            var act = new Func<Task>(async () => await service.ChangeOfParty(request));
+
+            act.Should().Throw<EntityNotFoundException<Domain.Entities.Reservation>>();
+        }
+
+        [Test, RecursiveMoqAutoData]
+        public void And_Reservation_Not_Confirmed_Status_Then_Throws_ArgumentException(
```

Comment:
Do we need a similar test to cover the scenario of making sure that we can't clone a reservation with a deleted status?

Reply:
done